### PR TITLE
Improve performance by fixing some type inference issues

### PR DESCRIPTION
**Pull request by MilesCranmer**
This gives a 2x speedup by fixing some type inference issues (found with Cthulhu.jl). In particular, some global variables were not declared constant (note that they can still be mutable even if declared constant; it simply declares the type as constant).

Before:
```julia
julia> @btime VimBindings.Parse.parse_command("d10w")
  6.717 μs (44 allocations: 1.59 KiB)
```

After:

```julia
julia> @btime VimBindings.Parse.parse_command("d10w")
  3.365 μs (35 allocations: 1.58 KiB)

### CODE REVIEW

**Comment by MilesCranmer** [comment]
I think it might even be a bigger change for functionality working with the `STATE` variable. Feels much quicker to switch modes.

**Comment by caleb-allen** [comment]
This is really great, exactly the type of work this package needs. Way too much code that was written just to "work" that has never been optimized or looked at for performance. Looking over it now.

I also just added some basic precompilation in 06a5093, I think together these changes will make for a really awesome improvement of the user experience

File: src/parse.jl [hunk 1]
```diff
@@ -7,7 +7,7 @@ import ..Util.log
 
 export well_formed, matched_rule, parse_command, synonym, partial_well_formed
 
-REGS = (
+const REGS = (
```
Comment (caleb-allen) [inline src/parse.jl:10]: ya know as a Java developer by trade I'm embarrassed to have so many un-`const`ed fields!

File: src/parse.jl [hunk 2]
```diff
@@ -139,12 +139,12 @@ function parse_command(s :: AbstractString) :: Union{Command, Nothing}
...
+    r::Regex
 
     m = match(r, s)
-    args = [ parse_value(capture) for capture in m.captures ]
-
-    command_type = rules[r]
-    command_type(args...)
+    m === nothing && return nothing
+    m::RegexMatch
+    return RULES[r](parse_value.(m.captures)...)
```
Comment (caleb-allen) [inline src/parse.jl:147]: very nice

**Comment by MilesCranmer** [comment]
Thanks!

There seems to be some issue on this PR where if you type a key that is not mapped (like `m` or `u`) in normal mode, the REPL will freeze.

On master when you do this, the REPL will temporarily freeze, but eventually returns control. I'm not sure which part of the code is getting hit when this happens.

**Review by caleb-allen — APPROVED** [review]
Looking good. I'm really excited about the speedups in here!

**Comment by caleb-allen** [comment]
> On master when you do this, the REPL will temporarily freeze, but eventually returns control. I'm not sure which part of the code is getting hit when this happens.

@MilesCranmer when this occurs, what happens when you press `Escape`? I've experienced freezing like this, where the command key stack accumulates more and more keys, trying (and failing) to parse it each time. `Escape` clears the stack.

Definitely bad behavior, I've started down a route to ensure that whenever an invalid vim command is typed the stack is cleared

https://github.com/caleb-allen/VimBindings.jl/blob/master/src/parse.jl#L71

Not quite done though

**Comment by caleb-allen** [comment]
Great! I'll try to get a release out with this in it in the next day or two

**Pull request merged**